src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -655,7 +665,20 @@ def _fix_pandas_df_integer(
             and column_metadata.scale == 0
             and not str(pandas_dtype).startswith("int")
         ):
-            pd_df[pandas_col_name] = pandas.to_numeric(
-                pd_df[pandas_col_name], downcast="integer"
-            )
+            # pandas.to_numeric raises "RuntimeWarning: invalid value encountered in cast"
+            # when it tried to downcast and loses precision. In this case, we try to convert to
+            # int64 using astype() and fallback to to_numeric if we were unsuccessful.
+            with warnings.catch_warnings(record=True) as warning_list:
+                pd_col_with_numeric_downcast = pandas.to_numeric(
+                    pd_df[pandas_col_name], downcast="integer"
+                )
```

Comment:
This isn't reliable but it won't make things worse. We can use it for now if no other options. Could you create a JIRA to follow up?

Reply:
We had a JIRA earlier about this issue. I am adding additional context to it and will write-up a proposal for GS/XP team to help us come-up with a better solution.

https://snowflakecomputing.atlassian.net/browse/SNOW-574649